extensions/vscode-api-tests/src/singlefolder-tests/proxy.test.ts

```diff
@@ -106,17 +106,26 @@ import { middleware, Straightforward } from 'straightforward';
 			const proxyPort = (sf.server.address() as AddressInfo).port;
 
 			await vscode.workspace.getConfiguration().update('integration-test.http.proxy', `PROXY 127.0.0.1:${proxyPort}`, vscode.ConfigurationTarget.Global);
-			await delay(1000); // Wait for the configuration change to propagate.
-			await new Promise<void>((resolve, reject) => {
-				https.get(url, res => {
-					if (res.statusCode === 418) {
-						resolve();
-					} else {
-						reject(new Error(`Unexpected status code (expected 418): ${res.statusCode}`));
+			for (let i = 0; i < 3; i++) {
+				try {
+					await new Promise<void>((resolve, reject) => {
+						https.get(url, res => {
+							if (res.statusCode === 418) {
+								resolve();
+							} else {
+								reject(new Error(`Unexpected status code (expected 418): ${res.statusCode}`));
+							}
+						})
+							.on('error', reject);
+					});
+					break; // Exit the loop if the request is successful
+				} catch (err) {
+					if (i === 2) {
+						throw err; // Rethrow the error if it's the last attempt
 					}
-				})
-					.on('error', reject);
-			});
+					await delay(2000); // Wait for the configuration change to propagate.
```

Comment:
There isn't a better way to wait for this?

Reply:
Not sure, trying here: https://github.com/microsoft/vscode/pull/237904